### ISSUE TITLE
I2I: Introducing denakop extension

### ISSUE THREAD

**Issue by gurodrigues** [ampproject/amphtml#22916]
## Summary
This code is meant to enable online content producers to have the freedom to quickly enable and disable advertising. From Denakop platform, it is possible to do it in a simple and intuitive way. The user may choose the advertising positions that will be activated and insert them on the page automatically. This is made possibile because this feature adapts the formats in a pre-mapped way and allows them to be handled later from a simple and user-friendly interface ideal to the end user.

## Motivation
The reason the functionality is being implemented is to allow pages to quickly enable and disable ads through Denakop's platform. It is a tool that allows online publishers to save time on ad implementation and rather stay focused on effectively publishing relevant content.

## Usage example
```
  <amp-denakop data-tag-id="2819896d-f724" data-publisher-id="22" layout="nodisplay"></amp-denakop>
  <amp-iframe width="1" title="User Sync"
              height="1"
              sandbox="allow-scripts"
              frameborder="0"
              src="https://cdn.jsdelivr.net/npm/prebid-universal-creative@latest/dist/load-cookie.html?endpoint=appnexus&max_sync_count=5">
    <amp-img layout="fill" src="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==" placeholder></amp-img>
  </amp-iframe>
```

/cc @ampproject/wg-approvers
/cc @lannka

**Comment by lannka** [comment]
@gurodrigues thanks for creating this.

could you help me understand what makes denakop different from a regular ad network? and in your example, what is the amp-iframe for?

**Comment by gurodrigues** [comment]
@lannka 

> could you help me understand what makes denakop different from a regular ad network?

Denakop is more than an adnetwork because we provide a entire ad serving and managing technology that enhances publishers operation in many aspects, such as:
. Direct campaigns and use of first party data
. Google's ad policy and coallition for better ads compliance
. Demand management
. Content verification
. User experience reports
. Survey system
. Signature system

Those and many other features makes us different from other ad networks in publisher's perspective. Altough we have caractheristics that relate us to ad networks, our bunisess proposal goes a lot further than just serving ads.

> and in your example, what is the amp-iframe for?

In our platform the publishers can add and/or remove new partners, such as: AppNexus, ROIx, Rubicon, Oath and many others. To do this competition between partners possible we use the AMP RTC and the prebid-server; and the iframe is necessary to be able to synchronize the cookies of all the partners.

**Comment by gurodrigues** [comment]
Hi @lannka,

How are you? 

I would like to know if you have anymore questions for us so we can continue this approval process.
Should I take any more step before you can approve it?

Thank you,
Gustavo Rodrigues

**Comment by lannka** [comment]
It's still unclear to me what you plan to do in your extension. 
> . Direct campaigns and use of first party data
. Google's ad policy and coallition for better ads compliance
. Demand management
. Content verification
. User experience reports
. Survey system
. Signature system

Do you plan just to load an iframe and run your own javascripts inside? Or do you plan to write all the features in this proposed extension? 

> quickly enable and disable ads through Denakop's platform. It is a tool that allows online publishers to save time on ad implementation and rather stay focused on effectively publishing relevant content.

How will this be implemented?

For the cookie syncing iframe
- should it be invisible? is the placeholder img necessary?
- does it communicate with amp-denakop in anyway?

In the meantime, please consider a much simpler integration approach (documented [here](https://github.com/ampproject/amphtml/blob/master/ads/README.md)) .

In general, a new extension has to be approved by @ampproject/wg-approvers . So if you think the existing `amp-ad` / `amp-embed` will not meet your need, please use this I2I as to communicate what you plan to implement in AMP.

**Comment by gurodrigues** [comment]
@lannka Thanks for the update and we look forward to the approval of your team. I'm not sure why I am not able to find them to add here @ampproject/wg-approvers.

> It's still unclear to me what you plan to do in your extension.

Denakop is an American company, headquartered in Miami with a strong presence in the Brazilian market. The company is also looking forward to conquer the American market.

Every day we come across the difficulty that publishers have with implementing new ad units on their pages and managing their direct campaigns. To solve this problem, we've developed a unique javascript that when installed is able to dynamically create ad units that can be configured through our platform. No technical knowledge is needed.

Within our platform, besides the publisher being able to create, activate and deactivate the ad units, it also manages to add new platforms to compete in that advertising space (header bidding). The platform also provides a number of other features, such as user experience reporting, Google policy violation reporting, coallition for better ads violations report, insertion of targeted campaigns, and more.

All changes made within our platform are automatically transmitted to the javascript installed on the Publisher page.

We have been operating in this market for four years and have more than sixthy publishers working with us. We are in the Google partner program for ADX and we want to expand our technology to AMP as well.


> Do you plan just to load an iframe and run your own javascripts inside? Or do you plan to write all the features in this proposed extension?

No, I do not plan to run my own javascript inside. All features will be implemented in the `amp-denakop` extention. This iframe is necessary to implement the Prebid Server and you can get more details in their page (http://prebid.org/formats/amp.html).
The iframe as I said is only used to properly sync user IDs with Prebid server, nothing more according to Prebid server for AMP documentation.

> How will this be implemented?

To implement our technology, you need to create an account with us and then install our tag just as described in the usage example.

> For the cookie syncing iframe should it be invisible? is the placeholder img necessary?

Yes, it would be invisible. And the placeholder is necessary to avoid the positioned iframe error.

> does it communicate with amp-denakop in anyway?

It will not communicate with `amp-denakop` directly but throw our Prebid server for user IDs sync, this is usually sent by our partners (Appneux, Rubicon, ROIx, AOL and others).


Unfortunately, we are not able to use the `amp-ad` / `amp-embed` as suggested, since they do not give us the flexibility to automate the implementation of the ad units dynamically, so little optimise and segment in an automated way.

**Comment by lannka** [comment]
> All features will be implemented in the amp-denakop extention.

@gurodrigues this will need a thorough discussion in this I2I or a separate design doc to cover all the implementation details. I'm especially interested in the automating ad units part. 

To avoid further back-n-forth here, we can connect on our [Slack channel](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md#discussion-channels).

**Comment by gurodrigues** [comment]
Hi @lannka,

We are glad to hear this and we will do our best to implement this technology as soon as possible.

I just sign up on Slack and i will send you a private message there.

Thank you

**Comment by gurodrigues** [comment]
Hi @lannka ,

As requested follows below the design doc:
https://drive.google.com/file/d/1sT80MW1i9Z1PkXycNkGFoBc5gaw57NPJ/view?usp=sharing

Thanks,

**Comment by lannka** [comment]
Thanks @gurodrigues for getting this out.

Since you're already on Google drive, do you mind move it to a Google doc so we can comment and discuss on it?

**Comment by gurodrigues** [comment]
@lannka I just updated the link above.

**Comment by lannka** [comment]
closed by #24404
